Sources/Fuzzilli/Core/ProgramBuilder.swift

```diff
@@ -302,13 +302,20 @@ public class ProgramBuilder {
         return randVar(ofType: type)
     }
 
+    /// Returns a random variable from the outer scope.
+    public func randVarFromOuterScope() -> Variable {
+        assert(hasVisibleVariables)
+        return randVarInternal(fromOuterScope: true)!
+    }
+
     /// Returns a random variable satisfying the given constraints or nil if none is found.
-    private func randVarInternal(_ selector: ((Variable) -> Bool)? = nil) -> Variable? {
+    private func randVarInternal(_ selector: ((Variable) -> Bool)? = nil, fromOuterScope: Bool = false) -> Variable? {
         var candidates = [Variable]()
+        let scopes = fromOuterScope && scopeAnalyzer.scopes.count > 1 ? scopeAnalyzer.scopes.dropLast() : scopeAnalyzer.scopes
 
         // Prefer inner scopes
         withProbability(0.75) {
-            candidates = chooseBiased(from: scopeAnalyzer.scopes, factor: 1.25)
+            candidates = chooseBiased(from: scopes, factor: 1.25)
             if let sel = selector {
                 candidates = candidates.filter(sel)
             }
```

Comment:
Don't you also need to handle lines 324 - 330? Probably you could just do something like `let visibleVariables = fromOuterScope ? scopes.reduce([], +) : scopeAnalyzer.visibleVariables` or so?